### PR TITLE
feat: utilise pytest for `test_approximation`

### LINKED ISSUE #532: Use pytest parameterization for `test_approximation.py`
### What's the new feature?

Utilise pytest parameterization and fixtures for the repeated test cases in `test_approximation.py`

### What value does this add?

Reduces the size of the tests, better highlights where the tests for specific implementations diverge, and reduces the opportunity for mistakes in/across the tests themselves.

### Is there an alternative you've considered?

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tc85324**
### PR Type
Closes #532

- Feature
- Refactoring (no functional changes)
- Tests

### Description
Refactors the tests in `test_approximation.py` to utilise pytest fixtures and parameterization.

### How Has This Been Tested?
The new tests pass.

### Does this PR introduce a breaking change?
Removes the `test_kernel_mean_approximator_creation` as this is implicitly tested by all the other tests.
Uses the same "almost_equal" tolerance for all approximators (pulls random down to the tolerance used in ANN and Nystrom).

### Checklist before requesting a review
- [x] I have made sure that my PR is not a duplicate.
- [x] My code follows the style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my code.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: tests/unit/test_approximation.py [hunk 1]
```diff
@@ -74,555 +79,228 @@ def setUp(self) -> None:
...
-            num_kernel_points=self.data.shape[0],
-            num_train_points=self.data.shape[0],
-        )
+        if not issubclass(approximator, coreax.approximation.NystromApproximator):
+            full_kwargs = {
+                "num_kernel_points": data.shape[0],
+                "num_train_points": data.shape[0],
+            }
+        else:
+            full_kwargs = {"num_kernel_points": data.shape[0]}
```
Comment (rg936672) [inline tests/unit/test_approximation.py:140]: My personal preference would be for these if-else blocks to be `if issubclass(...): ... else: ...`; that reads slightly clearer to me.

Comment (rg936672) [reply tests/unit/test_approximation.py:140]: (Same comment applies to the rest of the `if not issubclass` checks in this file!)

**Review by rg936672 — DISMISSED** [review]
This looks good - an impressive reduction in size! It's approaching the limit of how complex you can reasonably make a parametrised test before it's a better idea to just split it up, but as is it's still understandable. 

I've left one suggestion for a minor style change, but it's not critical. Other than that, this looks good to merge.

File: tests/unit/test_approximation.py (hunk 1 above)
Comment (tc85324) [reply tests/unit/test_approximation.py]: I agree and have updated accordingly

Comment (rg936672) [reply tests/unit/test_approximation.py]: Looks good to me!

**Review by rg936672 — APPROVED** [review]

**Pull request merged**